gateway/src/connectors/defira/defira.ts

```diff
@@ -192,26 +176,25 @@ export class Defira implements Uniswapish {
    * @param amount Amount of `baseToken` to put into the transaction
    */
   async estimateSellTrade(
-    baseToken: UniswapToken,
-    quoteToken: UniswapToken,
+    baseToken: DefiraToken,
+    quoteToken: DefiraToken,
     amount: BigNumber,
     allowedSlippage?: string
   ): Promise<ExpectedTrade> {
-    const nativeTokenAmount: UniswapTokenAmount = new UniswapTokenAmount(
-      baseToken,
-      amount.toString()
-    );
+    const nativeTokenAmount = DefiraTokenAmount.fromRawAmount(baseToken, amount.toString());
+
     logger.info(
       `Fetching pair data for ${baseToken.address}-${quoteToken.address}.`
     );
 
     // TODO: replace fetcher which returns defira-sdk Pair instead of this uniswap-sdk Pair
-    const pair: UniswapPair = await UniswapFetcher.fetchPairData(
+    const pair: DefiraPair = await DefiraFetcher.fetchPairData(
       quoteToken,
       baseToken,
-      this.harmony.provider
+      this._factory,
+      "INIT_HASH_PLACEHOLDER"
```

Comment:
Probably good to add to the [config](https://github.com/NavneethJayendran/hummingbot/blob/defira-connector/gateway/src/connectors/defira/defira.config.ts).

Reply:
Still not addressed